### PR TITLE
Upgrade to Rust 1.64.0

### PR DESCRIPTION
**Pull request by brooksprumo**
As the title says 😊

### CODE REVIEW

File: banks-client/src/lib.rs [hunk 1]
```diff
@@ -525,6 +525,7 @@ mod tests {
     }
 
     #[test]
+    #[allow(clippy::result_large_err)]
```
Comment (t-nelson) [inline banks-client/src/lib.rs:528]: interesting that this lint missed all the instances where we return `Result<(), BankClientError>` via `Future` :thinking:

**Review by t-nelson — COMMENTED** [review]
lgtm. thanks!

**Review by steviez — APPROVED** [review]
LGTM

**Review by brennanwatt — APPROVED** [review]
LGTM

**Pull request merged**